profile/profile.js

```diff
@@ -98,12 +104,67 @@ WDN.initializePlugin('modal', [function() {
 				}
 			});
 		}
+
+
+
+		// Maps a roleId -> siteId -> maximumNumber of that role allowed for that site (shift can override)
+		const siteRoleLimitsMap = new Map();
+		// Maps a roleId -> shiftId -> maximumNumber of that role allowed for that shift
+		const shiftRoleLimitsMap = new Map();
+
+		function loadRoleLimits() {
+			$.ajax({
+				url: '/server/api/roles/limits/getAll.php',
+				type: 'GET',
+				dataType: 'JSON',
+				cache: false,
+				success: function(response) {
```

Comment:
The `success` and `error` methods are deprecated and are removed in jQuery 1.8. I would recommend using `AJAX.done` and `AJAX.error` if you want to upgrade jQuery in the future. I don't believe there is a performance difference.

Reply:
Going off this, have you considered making generic Ajax methods that abstract some of this boiler-plate out of the way? Usually they're helpful for generic error tracking and if you run into issues like this where parts of jquery is depreciated, you'd only have to change it in one place.

Reply:
It's never crossed my mind, but we should definitely be doing this...